src/main/java/bio/terra/service/dataset/DatasetRelationshipDao.java

```diff
@@ -52,32 +55,23 @@ protected void create(Relationship relationship) {
   }
 
   public void retrieve(Dataset dataset) {
-    List<UUID> columnIds = new ArrayList<>();
-    dataset
-        .getTables()
-        .forEach(table -> table.getColumns().forEach(column -> columnIds.add(column.getId())));
-    dataset.relationships(
-        retrieveDatasetRelationships(
-            columnIds, dataset.getTablesById(), dataset.getAllColumnsById()));
+    List<Relationship> relationships = retrieveDatasetRelationships(dataset);
+    dataset.relationships(relationships);
   }
 
-  private List<Relationship> retrieveDatasetRelationships(
-      List<UUID> columnIds, Map<UUID, DatasetTable> tables, Map<UUID, Column> columns) {
-    String sql =
-        "SELECT id, name, from_table, from_column, to_table, to_column "
-            + "FROM dataset_relationship WHERE from_column IN (:columns) OR to_column IN (:columns)";
-    MapSqlParameterSource params = new MapSqlParameterSource().addValue("columns", columnIds);
-    return jdbcTemplate.query(
-        sql,
-        params,
-        (rs, rowNum) ->
-            new Relationship()
-                .id(rs.getObject("id", UUID.class))
-                .name(rs.getString("name"))
-                .fromTable(tables.get(rs.getObject("from_table", UUID.class)))
-                .fromColumn(columns.get(rs.getObject("from_column", UUID.class)))
-                .toTable(tables.get(rs.getObject("to_table", UUID.class)))
-                .toColumn(columns.get(rs.getObject("to_column", UUID.class))));
+  private List<Relationship> retrieveDatasetRelationships(Dataset dataset) {
+    String select =
+        """
+            SELECT r.id, r.name, r.from_table, r.from_column, r.to_table, r.to_column
+            FROM dataset_relationship AS r
+            -- Assumption: from_table and to_table will be in the same dataset, saving us a join.
```

Comment:
Good idea -- let me chew on how to best achieve this (but am very open to suggestions).

Reply:
I think to achieve this I will refactor our public relationship creation method to take in a `Dataset`, rather than arbitrary `Relationship`s.  Then that method will extract the relationship specifications from a single dataset.

Reply:
Hmm, unfortunately my initial idea isn't neatly compatible with our support for adding new relationships to a dataset after its creation.

Here are a few attainable options:
-  Expand docstrings to emphasize this expectation (does not give us coverage of this assumption in tests, however).
- Expand relationship creation method to take in a dataset ID, but not use it anywhere :( only to signal to its callers that a relationship is attributed to a single dataset.

For completeness, the most thorough approach would be:
- Add a new `dataset_id` column to `dataset_relationship`.
- Write `dataset_id` when creating new `dataset_relationship`s.
- Add an admin upgrade flight to back-populate `dataset_id`s for all existing relationships.
- Simplify relationship retrieval even further: no join required.

(I don't think that thorough approach is worth it, relative to the low likelihood of risk and competing demands for our time.)

@fboulnois , what do you think?

Reply:
Given the complexity I vote for the docstrings approach, which seems the most bang for buck.

Reply:
Docstrings added -- thanks!